File: fsharp/armstrong-numbers/ArmstrongNumbersTests.fs
module ArmstrongNumbersTests

open FsUnit.Xunit
open Xunit

open ArmstrongNumbers

[<Fact>]
let ``Zero is an Armstrong number`` () =
    isArmstrongNumber 0 |> should equal true

[<Fact>]
let ``Single digit numbers are Armstrong numbers`` () =
    isArmstrongNumber 5 |> should equal true

[<Fact>]
let ``There are no 2 digit Armstrong numbers`` () =
    isArmstrongNumber 10 |> should equal false

[<Fact>]
let ``Three digit number that is an Armstrong number`` () =
    isArmstrongNumber 153 |> should equal true

[<Fact>]
let ``Three digit number that is not an Armstrong number`` () =
    isArmstrongNumber 100 |> should equal false

[<Fact>]
let ``Four digit number that is an Armstrong number`` () =
    isArmstrongNumber 9474 |> should equal true

[<Fact>]
let ``Four digit number that is not an Armstrong number`` () =
    isArmstrongNumber 9475 |> should equal false

[<Fact>]
let ``Seven digit number that is an Armstrong number`` () =
    isArmstrongNumber 9926315 |> should equal true

[<Fact>]
let ``Seven digit number that is not an Armstrong number`` () =
    isArmstrongNumber 9926314 |> should equal false

File: csharp/armstrong-numbers/ArmstrongNumbersTests.cs
using Xunit;

public class ArmstrongNumbersTests
{
    [Fact]
    public void Zero_is_an_armstrong_number()
    {
        Assert.True(ArmstrongNumbers.IsArmstrongNumber(0));
    }

    [Fact]
    public void Single_digit_numbers_are_armstrong_numbers()
    {
        Assert.True(ArmstrongNumbers.IsArmstrongNumber(5));
    }

    [Fact]
    public void There_are_no_two_digit_armstrong_numbers()
    {
        Assert.False(ArmstrongNumbers.IsArmstrongNumber(10));
    }

    [Fact]
    public void Three_digit_number_that_is_an_armstrong_number()
    {
        Assert.True(ArmstrongNumbers.IsArmstrongNumber(153));
    }

    [Fact]
    public void Three_digit_number_that_is_not_an_armstrong_number()
    {
        Assert.False(ArmstrongNumbers.IsArmstrongNumber(100));
    }

    [Fact]
    public void Four_digit_number_that_is_an_armstrong_number()
    {
        Assert.True(ArmstrongNumbers.IsArmstrongNumber(9474));
    }

    [Fact]
    public void Four_digit_number_that_is_not_an_armstrong_number()
    {
        Assert.False(ArmstrongNumbers.IsArmstrongNumber(9475));
    }

    [Fact]
    public void Seven_digit_number_that_is_an_armstrong_number()
    {
        Assert.True(ArmstrongNumbers.IsArmstrongNumber(9926315));
    }

    [Fact]
    public void Seven_digit_number_that_is_not_an_armstrong_number()
    {
        Assert.False(ArmstrongNumbers.IsArmstrongNumber(9926314));
    }
}